debezium-core/src/test/java/io/debezium/transforms/FilterTest.java

```diff
@@ -0,0 +1,161 @@
+ */
+package io.debezium.transforms;
+
+import java.time.Instant;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.connect.data.SchemaBuilder;
+import org.apache.kafka.connect.data.Struct;
+import org.apache.kafka.connect.source.SourceRecord;
+import org.fest.assertions.Assertions;
+import org.junit.Test;
+
+import io.debezium.DebeziumException;
+import io.debezium.data.Envelope;
+
+/**
+ * @author Jiri Pechanec
+ */
+public class FilterTest {
+
+    private static final String LANGUAGE = "language";
+    private static final String EXPRESSION = "condition";
+    private static final String NULL_HANDLING = "null.handling.mode";
+
+    final Schema recordSchema = SchemaBuilder.struct()
+            .field("id", SchemaBuilder.int8())
+            .field("name", SchemaBuilder.string())
+            .build();
+
+    final Schema sourceSchema = SchemaBuilder.struct()
+            .field("lsn", SchemaBuilder.int32())
+            .build();
+
+    final Envelope envelope = Envelope.defineSchema()
+            .withName("dummy.Envelope")
+            .withRecord(recordSchema)
+            .withSource(sourceSchema)
+            .build();
+
+    @Test(expected = DebeziumException.class)
+    public void testLanguageRequired() {
+        try (final Filter<SourceRecord> transform = new Filter<>()) {
+            final Map<String, String> props = new HashMap<>();
+            props.put(EXPRESSION, "operation == 'd'");
+            transform.configure(props);
+        }
+    }
+
+    @Test(expected = DebeziumException.class)
+    public void testExpressionRequired() {
+        try (final Filter<SourceRecord> transform = new Filter<>()) {
+            final Map<String, String> props = new HashMap<>();
+            props.put(LANGUAGE, "groovy");
+            transform.configure(props);
+        }
+    }
+
+    @Test(expected = DebeziumException.class)
+    public void shouldFailOnUnkownLanguage() {
+        try (final Filter<SourceRecord> transform = new Filter<>()) {
+            final Map<String, String> props = new HashMap<>();
+            props.put(EXPRESSION, "operation == 'd'");
+            props.put(LANGUAGE, "jython");
+            transform.configure(props);
+        }
+    }
+
+    @Test(expected = DebeziumException.class)
+    public void shouldFailToParseCondition() {
+        try (final Filter<SourceRecord> transform = new Filter<>()) {
+            final Map<String, String> props = new HashMap<>();
+            props.put(EXPRESSION, "operation == 'd");
+            props.put(LANGUAGE, "groovy");
+            transform.configure(props);
+        }
+    }
+
+    @Test
+    public void shouldProcessCondition() {
+        try (final Filter<SourceRecord> transform = new Filter<>()) {
+            final Map<String, String> props = new HashMap<>();
+            props.put(EXPRESSION, "value.op == 'd' && value.before.id == 2");
```

Comment:
Very nice! What I didn't get yet: how does evaluation happen? I saw you bind key, value etc. into the Groovy execution context. But how is the `before.id` part evaluated?

Reply:
@gunnarmorling That's Groovy magic :-), if such properties are not available but method `get` is  then it is used - that's derive from Map handling.

Reply:
Ah, I see. So they automagically resolve it to `.get("before").get("id")`. Interesting indeed :)